### ISSUE TITLE
about the PCA on the training set

### ISSUE THREAD

**Issue by zrqi** [justusschock/shapenet#11]
Hi,I read the paper on shapenet, but I have a problem, that is, when you perform PCA  of landmarks on the training set, the shape of all samples is used as a whole set to perform PCA  and then take the eigenvector. Is it? If so, how do you find the weight of each eigenvector for each sample?

**Comment by justusschock** [comment]
I don't really understand your problem. The basic pipeline is:

1. Create a PCA on the trainset's landmarks
2. Use this PCA (only the first N compontens + mean) inside the shape model
3. Learn the weighting parameters and global transformation parameters by CNN

An in prediction:
1. Use trained CNN to predict parameters and apply PCA-Layer and global transformation to obtain shapes

**Comment by zrqi** [comment]
I mainly want to ask what is the label when training the cnn? Is it the weight coefficient ？How to calculate it based on the training sample

**Comment by justusschock** [comment]
No, the label are the coordinates of the groundtruth points. We use a simple distance loss (L1) between the predicted point coordinates and the groundtruth coordinates, which is implicitly mapped on the weight coefficients by the proposed layer.

**Comment by zrqi** [comment]
Thank you very much, I understand.

**Comment by mohaEs** [comment]
In your experimental environment,  you have 68 landmarks for each face and you set the PCA components to 25 in the config file. Based on your experience, what would be a suitable choice of PCA components if we have just 23 landmarks in each image?
